lib/AST/Module.cpp

```diff
@@ -2623,8 +2624,17 @@ ModuleLibraryLevelRequest::evaluate(Evaluator &evaluator,
 }
 
 bool SourceFile::shouldCrossImport() const {
-  return Kind != SourceFileKind::SIL && Kind != SourceFileKind::Interface &&
-         getASTContext().LangOpts.EnableCrossImportOverlays;
+  if (Kind == SourceFileKind::SIL || Kind == SourceFileKind::Interface)
+    return false;
+  // Cross import when experimental differentiable programming is enabled. This
+  // is to enable tgmath derivatives defined in the platform differentiation
+  // overlay.
+// TODO(philipturner) Try enabling this once
+//   SwiftCompilerSources/Source/ExperimentalRegex isn't broken
+  
+//  if (isDifferentiableProgrammingEnabled(*this))
+//    return true;
+  return getASTContext().LangOpts.EnableCrossImportOverlays;
 }
```

Comment:
Why do we need to change this method?

Reply:
That's how we're going to actually support cross-import overlays. I'm doing the best I can right now because a compiler crash stops me from testing whether this actually enables cross-import overlays or not.

Reply:
@rxwei I saw that you made some pull requests about experimental string processing. If that's what your current focus is, would you be able to investigate this build failure?

Reply:
Sorry I do not have any bandwidth to investigate this.

Reply:
I removed it, but I'm still keeping the new line break as a nice cleanup of the existing code.